docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+
+We start with the simplest possible case that occurs in `binarySearch`. Namely,
+we consider the case of `low > high`, that is, `binarySearch` is never entering
+the loop.
+
+**Introduce variables**. To do that, we have to finally introduce some
+variables. Obviously, we have to introduce variables `low` and `high`. This is
+how we do it:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:33:38}}
+```
+
+The variables `low` and `high` are called *state variables*. They are always
+present in a state of our state machine. That is, they are never introduced and
+never removed. Remember, TLA+ is not tuned towards any particular computer
+architecture and thus it does not even have the notion of an execution stack.
+You can think of `low` and `high` as being global variables. Yes, global
+variables are generally a bad smell in a program. However, when dealing with a
+specification, they are much easier to reason about than the execution stack.
+If you need to introduce local definitions, you will see how to do that later
+in this tutorial.
+
+Less obviously, we introduce two additional variables:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch2.tla:39:44}}
+```
+
+The variable `isTerminated` indicates, whether our search has terminated. Why
+did we even have to introduce it? Because, in general, computer systems do not
+have to terminate. If we want to specify the Internet or Bitcoin, do we
+understand what it means for them to terminate?
+
+The variable `returnValue` will contain the result of the binary search, when
+the search terminated. Recall, there is no execution stack. Hence, we introduce
+the variable `returnValue` right away. The downside is that we have to do
+book-keeping for this variable.
+
+**Initialize variables.** Having introduced the variables, we have to
+initialize them. That is, we want to specify lines 2-3 of the Java code:
+
+```java
+1:     public static int binarySearch(int[] a, int key) {
+2:         int low = 0;
+3:         int high = a.length - 1;
+           ...
+17:    }
+```
+
+To this end, we change the body of the predicate `Init` to the following:
+
+```tla
+Init ==
+  low = 0 /\ high = Len(INPUT_SEQ) - 1 /\ isTerminated = FALSE /\ returnValue = 0
+```
+
+You probably have guessed, what the above line means. You are probably a bit
+puzzled about the mountain-like operator `/\`. It is called *conjunction*,
+which is usually written as `&&` or `and` in programming languages. The
+important effect of the above expression is that every variable in the
+left-hand side of `=` is required to have the value specified in the right-hand
+side of `=`. It is important to know that TLA+ does not impose any particular
+order of evaluation for `/\`. However, both Apalache and TLC evaluate
+some expressions of the form `x = e` in the initialization predicate as
+assignments. Hence, it is often a good idea to think about `/\` as being
+evaluated from left to right.
```

Comment:
This might be too technical at this point. Or maybe we could make it into a sidenote in a box or something?